classes/task.class.php

```diff
@@ -166,6 +166,7 @@ public function get_reports() {
             // webhook callback so ignore anything submitted to Turnitin in the 2 minutes.
             // Otherwise retrieve originality score if we haven't received it back within 5 minutes.
             if ($tssubmission->getstatus() == TURNITINSIM_SUBMISSION_STATUS_UPLOADED
+                && $tssubmission->get_submission_info()->status === TURNITINSIM_SUBMISSION_STATUS_COMPLETE
```

Comment:
Nice addition here!

Reply:
Yeah currently there is no existing unit test and I believe because it involves database mock that's the reason there is no unit test.